src/myrtlespeech/run/train.py

```diff
@@ -47,46 +47,56 @@ def fit(
 
```

Comment:
Keeping the interface limited to one function is still somewhat appealing as it reduces the size of the API exposed to the user. It is possible to keep the single `fit` function interface by doing the following:

```python
def fit(
    seq_to_seq: SeqToSeq,
    epochs: int,
    train_loader: Optional[DataLoader] = None,
    eval_loader: Optional[DataLoader] = None,
    callbacks: Optional[Collection[Callback]] = None,
) -> None:

...

    for epoch in range(epochs):
        stages = []
        if train_loader is not None:
            if epoch == 0 and eval_loader is not None:
                stages.append(Stage.EVAL)   # eval before training starts
            stages.append(Stage.TRAIN)

        if eval_loader is not None:
            stages.append(Stage.EVAL)  # eval after every epoch
```

This means the `eval` and `run_stage` functions can be removed.

Thoughts?

Reply:
You're right - the eval wasn't adding much. I've followed the above suggestion and added a few checks to ensure that the passed args are sensible.